client/src/components/target/target.js

```diff
@@ -1,195 +1,251 @@
 import React from "react";
-import target from "./target.png";
 
 export default class Target extends React.Component {
   constructor(props) {
     super(props);
-    this.setCanvasRef = element => {
-      this.canvas = element;
-    };
+    this.canvasRef = React.createRef();
   }
 
-  componentDidMount() {
-    this.initTarget();
+  componentDidUpdate() {
+    this.context = this.canvasRef.current.getContext("2d");
+    this.context.clearRect(0, 0, this.canvasSizePx, this.canvasSizePx);
+    this.renderTargetAndArrows();
   }
 
-  componentDidUpdate() {
-    this.context = this.canvas.getContext("2d");
-    this.doCanvas();
+  componentDidMount() {
+    this.context = this.canvasRef.current.getContext("2d");
+    this.context.clearRect(0, 0, this.canvasSizePx, this.canvasSizePx);
+    this.context.resetTransform();
+    this.renderTargetAndArrows();
   }
 
   render() {
-    return <canvas ref={this.setCanvasRef} width="300" height="300" />;
+    return (
+      <canvas
+        ref={this.canvasRef}
+        width={this.canvasSizePx}
+        height={this.canvasSizePx}
+        onMouseEnter={() => {
```

Comment:
This is better than previous this.canvas.addEventListerner:
- less code to write
- don't need to removeEventListener on unmounting